Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/search.js

```diff
@@ -0,0 +1,153 @@
+            break;
+        }
+    }
+
+    if (VALID_REF_RE.test(commitRep) || !commitRep) {
+        message.innerHTML = '';
+        document.getElementById(this.id).classList.remove('invalid');
+    } else {
+        message.innerHTML = `<div class="text small failed">${commitRep} is not a valid commit format</div>`;
+        document.getElementById(this.id).classList.add('invalid');
+    }
+}
+
+async function GenerateCompareLink() {
+    const commitOne = document.getElementById('commit1').value;
+    const commitTwo = document.getElementById('commit2').value;
+    let message = document.getElementById('compareMessage');
+
+    if (!commitOne | !commitTwo) {
+        message.innerHTML = ERROR_TEMPLATE.replace("{message}", "Must provide two commits to compare");
+    } else if (!VALID_REF_RE.test(commitOne)) {
+        message.innerHTML = ERROR_TEMPLATE.replace("{message}", `${commitOne} is not a valid commit format<br>`);
+    } else if (!VALID_REF_RE.test(commitTwo)) {
+        message.innerHTML = ERROR_TEMPLATE.replace("{message}", `${commitTwo} is not a valid commit format<br>`);
+    } else {
+        try {
+            const commitOneData = await GetCommit(commitOne);
+            const commitTwoData = await GetCommit(commitTwo);
+            return COMMIT_URL + 'compare/' + commitOneData.identifier + '...' + commitTwoData.identifier;
+        } catch (error) {
+            console.error(error);
+            message.innerHTML = ERROR_TEMPLATE.replace("{message}", `${error.message}<br>`);
+        }
+    }
+    return;
+}
+
+async function CopyCompareLink() {
+    // Creating a Promise as a workaround for https://bugs.webkit.org/show_bug.cgi?id=222262
+    navigator.clipboard.write([
+        new ClipboardItem({
+            "text/plain": new Promise(async resolve => {
+                const compareLink = await GenerateCompareLink();
+                if (compareLink) {
+                    let message = document.getElementById('compareMessage');
+                    message.innerHTML = `<a href="${compareLink}"> ${compareLink} </a> copied to clipboard!`;
+                    resolve(compareLink);
+                }
+            })
+        })
+    ]);
+}
+
+async function CompareRedirect() {
+    let compareLink = await GenerateCompareLink();
+    if (compareLink) {
+        window.location.href = compareLink;
+    }
+}
+
+function FindCommit() {
+    const commit = document.getElementById('commitSearch').value;
+    let message = document.getElementById('searchMessage');
+    if (!commit) {
+        message.innerHTML = ERROR_TEMPLATE.replace("{message}", `Must provide a commit identifier<br>`);
+        return;
+    }
+
+    fetch(`/${commit}/json`).then(response => {
+        if (!response.ok) {
+            message.innerHTML = ERROR_TEMPLATE.replace("{message}", `Could not find ${commit}.<br>Please fix any typos or check if this commit exists.<br>`);
+        } else {
+            response.json().then(data => {
```

Comment:
We should handle the request error

Reply:
response.json() may also have an error such like your API somehow didn't give a json back, and we should handle this as well